students/perrymant/blog-01-hello-world-deconstructed.md

```diff
@@ -0,0 +1,347 @@
+            └── example                     <- package
+                ├── appliances              <- package
+                │   ├── Oven.java
+                │   └── Refrigerator.java
+                ├── food                    <- package
+                │   ├── Antipasti.java
+                │   ├── Pasta.java
+                │   ├── Pizza.java
+                │   └── ingredients         <- package
+                │       ├── Cheese.java
+                │       ├── Flour.java
+                │       └── Tomato.java
+                └── restaurant              <- package
+                    ├── Kitchen.java
+                    └── Toilet.java
+```
+Packages are used for organising related classes, and to put a class in a package, you need to add the line
+
+`package `*packagename*`;`
+
+as the very first instruction in the source file.
+Typically you will see these in a format that has one or more identifiers separated by periods such as `java.util.random`.
+The naming format here is trying to make sure that each name is unique, so as to avoid namespace collisions and is the same as with import naming.
+For example, if I wanted to make use of the class `List`, I would find that there are quite a few:
+![Namespace](assets/blog-01-hello-world-deconstructed/namespace.png)
+*which one do I want?* I have to specify, and this reversed naming format helps keep all of these classes and packages separate.
+
+When the `protected` access modifier is used, it signals that the method, constructors or data member is accessible both within and outside the package but through inheritance only.
+
+#### 3. `package private` access modifier
+
+When no named package statements are included in a source file, then the classes will be placed in the *default package*.
+The default modifier is accessible only within package.
+
+```java
+    ├── food                    <- package
+    │   ├── Antipasti.java
+    │   ├── Pasta.java
+    │   ├── Pizza.java
+    │   └── ingredients         <- package
+    │       ├── Cheese.java
+    │       ├── Flour.java
+    │       └── Tomato.java
+```
+
+If all the classes in the above situation were given the `package private` access modifier, then the class, field, method or constructor of the food package (Antipasti, Pasta and Pizza classe) would have no accessiblity to the class, field, method or constructor of the ingredients package (Cheese, Flour and Tomato classes).
+
+#### 4. `private` access modifier
+
+The `private` access modifier gives us the opportunity to permit or deny the linking of low-level components with each other, and is more often found on fields and methods, rather than on classes.
+A field or method that is marked as private will only allow code inside the same class to access the field, or call the method - it keeps everything very contained in the class.
+This means that code inside subclasses or external classes cannot access the fields or methods that are designated as `private`.
+
+```java
```

Comment:
I wholeheartedly approve of this message...

Is what I would like to say but the getter and setter kind of comes out of the blue. Maybe change the code to look more like a private field `private String intentions = "make collusion great again";`, and a public method:

```java
public String getIntentions() {
    return "NO COLLUSION!!!!!";
}
```

I think this way the encapsulation is more obvious.